### PR TITLE
[MDS-413] Fixed Modal v2, added animations

### PR DESCRIPTION
**Pull request by tkullisaar**
- Updated component anatomy (removed Modal.Header)
- Fixed examples
- Added animations

### CODE REVIEW

**Review by margusp2heathmont — APPROVED** [review]
Super! :)

File: assets/js/hooks/modal.js
Comment (phgrey) [inline assets/js/hooks/modal.js:1]: would like to take a deeper look here, comparing to hooks drawer.js & snackbar.js
maybe this is a single hook.
will talk on Maonday

File: lib/moon/design/modal.ex [hunk 1]
```diff
@@ -1,33 +1,56 @@
 defmodule Moon.Design.Modal do
   @moduledoc false
-  use Moon.StatelessComponent
+  use Moon.StatefulComponent
+
+  data(is_open, :boolean, default: false)
+  data(is_closing, :boolean, default: false)
```
Comment (phgrey) [inline lib/moon/design/modal.ex:6]: maybe it should be prop, so user can make it visible by default. the same for a drawer

File: lib/moon/design/modal.ex [hunk 2]
```diff
@@ -1,33 +1,56 @@
...
 
   slot(backdrop, required: true)
   slot(panel, required: true)
 
-  prop(is_open, :boolean, default: false)
-  prop(on_close, :event)
```
Comment (phgrey) [inline lib/moon/design/modal.ex:9]: leave it back, let programemrs to handle it by they own if they want

**Review by phgrey — COMMENTED** [review]
but see comments about is_open to be prop, not a data and about a close event

**Pull request merged**